### PR TITLE
add common insecure flag

### PR DESCRIPTION
**Pull request by jeff-mccoy**
Migrated from #1191

## Description
Adding support for insecure flag for external registry in `zarf init`. Fixing `zarf package create` and `zarf package deploy` failures with external self signed cert registry as mentioned in https://github.com/defenseunicorns/zarf/issues/1190 and https://github.com/defenseunicorns/zarf/issues/1181

## Related Issue

Fixes # https://github.com/defenseunicorns/zarf/issues/1190
Fixes # https://github.com/defenseunicorns/zarf/issues/1181
Fixes # https://github.com/defenseunicorns/zarf/issues/561
Fixes # https://github.com/defenseunicorns/zarf/issues/1015

### CODE REVIEW

File: src/cmd/root.go [hunk 1]
```diff
@@ -80,6 +81,7 @@ func init() {
 	rootCmd.PersistentFlags().BoolVar(&message.NoProgress, "no-progress", v.GetBool(V_NO_PROGRESS), lang.RootCmdFlagNoProgress)
 	rootCmd.PersistentFlags().StringVar(&config.CommonOptions.CachePath, "zarf-cache", v.GetString(V_ZARF_CACHE), lang.RootCmdFlagCachePath)
 	rootCmd.PersistentFlags().StringVar(&config.CommonOptions.TempDirectory, "tmpdir", v.GetString(V_TMP_DIR), lang.RootCmdFlagTempDir)
+	rootCmd.PersistentFlags().BoolVar(&config.CommonOptions.Insecure, "insecure", v.GetBool(V_INSECURE), lang.RootCmdFlagInseure)
```
Comment (Racer159) [inline src/cmd/root.go:84]: Can we add a `-k` alias like we had discussed earlier in the week to signal to folks this more closely follows `curl`s paradigms? 

```suggestion
	rootCmd.PersistentFlags().BoolVarP(&config.CommonOptions.Insecure, "insecure", "k", v.GetBool(V_INSECURE), lang.RootCmdFlagInseure)
```

**Review by Racer159 — CHANGES_REQUESTED** [review]
One small thing we had talked about earlier

File: src/config/lang/english.go [hunk 2]
```diff
@@ -37,6 +37,7 @@ const (
 	RootCmdFlagNoProgress  = "Disable fancy UI progress bars, spinners, logos, etc"
 	RootCmdFlagCachePath   = "Specify the location of the Zarf cache directory"
 	RootCmdFlagTempDir     = "Specify the temporary directory to use for intermediate files"
+	RootCmdFlagInseure     = "Allow access to insecure registries and disable other recommended security enforcements. This flag should only be used if you have a specific reason and accept the reduced security posture."
```
Comment (Racer159) [inline src/config/lang/english.go:40]: Not calling this out in the requested changes, but wanted to add a comment for @Madeline-UX to look at this verbiage to make sure it makes sense. `--insecure` here is doing two things:

1. Allowing connections to `http` image registries on `create` or `deploy`
2. Disabling Transport Layer Security (TLS) verification for connections to `https` registries

Technically both of these apply only to image registries at the moment and I do feel this message may require more balancing between `create` personas (who know what http/tls are) and `deploy` personas who may not.

Comment (jeff-mccoy) [reply src/config/lang/english.go:40]: Yeah I agree, I just left it generic as "other security enforcements" because of that.

**Review by Racer159 — APPROVED** [review]
lgtm

File: src/config/lang/english.go (hunk 2 above)
Comment (Madeline-UX) [reply src/config/lang/english.go:40]: Would it be better to specify which security enforcements it disables (the TLS)? That way, it's specific and can be researched if needed. 

I am concerned that the vagueness of "and other security enforcements" may keep users from being able to accurately assess the risk.  Unless we are going for scary - vague. Don't do this ... unless you really need to.

Comment (Madeline-UX) [reply src/config/lang/english.go:40]: @Racer159 @jeff-mccoy I could do a quick async UX test of the language - I need to know what we are wanting to test though. 

We assume that the creator understands and knows what these terms mean
We are not sure if the deploy persona "Ashton does?"

**Review by YrrepNoj — APPROVED** [review]

**Review by Racer159 — APPROVED** [review]
lgtm (again)

**Review by YrrepNoj — DISMISSED** [review]

**Review by YrrepNoj — APPROVED** [review]

**Review by Racer159 — APPROVED** [review]
sad pandas on -k, but lgtm

**Pull request merged**